src/Components/Auction/PostRegistrationModal.tsx

```diff
@@ -0,0 +1,111 @@
+import { Box, Button, CheckCircleIcon, Modal, Serif } from "@artsy/palette"
+import React, { useEffect, useState } from "react"
+
+export type ContentKey =
+  | "registrationConfirmed"
+  | "registrationPending"
+  | "bidPending"
+
+interface Props {
+  onClose: () => void
+  contentKey: ContentKey
+}
+
+type ModalContent = React.ComponentType<{ onClick: () => void }>
```

Comment:
Should this `React.FC`?

Reply:
Probably- just didn't want to limit myself [and IIRC autocomplete gave me that one first].